lasagne/layers/dnn.py

```diff
@@ -9,15 +8,16 @@
 from .pool import pool_output_length
 from ..utils import as_tuple
 
-if not theano.sandbox.cuda.cuda_enabled:
+if theano.sandbox.cuda.dnn.dnn_available():
+    from theano.sandbox.cuda import dnn
+    print("Using Old Theano Backend")     
+elif theano.sandbox.gpuarray.dnn.dnn_present():
```

Comment:
The latest version of Theano has moved gpuarray out of the sandbox, so this should just be`theano.gpuarray.dnn.dnn_present()` (and likewise for the rest) for those who are trying to use this PR with the dev version of Theano.